sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -598,6 +598,13 @@ fromClause
     : FROM relation (',' relation)* lateralView* pivotClause?
     ;
 
+temporalClause
+    : VERSION AS OF version=INTEGER_VALUE
```

Comment:
Wouldn't it be easier to combine the rules?

```
     : FOR? (SYSTEM_VERSION | VERSION) AS OF version=STRING
```

Otherwise, you have to remember that `SYSTEM_VERSION` requires `FOR` and `VERSION` is incompatible with it. That's a poor user experience because `FOR VERSION AS OF` would result in a parse error. Probably best to make `FOR` optional in both cases. Same for `TIMESTAMP` and `SYSTEM_TIME`

Reply:
@rdblue Changed. Thanks for the suggestion.

Reply:
@rdblue I looked hive syntax and it seems this `FOR` is always required for `SYSTEM_XXX`. Also checked SQL 2011 Standard https://en.wikipedia.org/wiki/SQL:2011#Temporal_support, SAP HANA uses `FOR SYSTEM_TIME AS OF` and IBM DB2 uses `FOR SYSTEM_TIME AS OF`. I removed the ? for `FOR` and made it only apply for `SYSTEM_XXX`.  Hope this is fine with you.